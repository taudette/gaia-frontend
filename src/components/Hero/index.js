import React from 'react';
import styles from '../../styles/components/_hero.scss'

const HeroComponent = (props) => {
  return (
    <section >
      <div className={styles.heroContainer}>
        <img src={props.hero.lgImage} alt="hero-image" />
        <div className={styles.text}>
          <h2>{props.hero.name}</h2>
          <p className="hidden-xs"><small>{props.hero.body}</small></p>
        </div>
      </div>
    </section>
  )
}

export default HeroComponent


