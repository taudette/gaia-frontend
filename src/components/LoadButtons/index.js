import React from 'react';
import styles from '../../styles/components/_loadButtons.scss'
import { Row, Col } from 'react-bootstrap'

const LoadButtonsComponent = (props) => {
  return (
    <Row className={styles.buttonRow}>
      {/* don't show load more button if no more to load*/}
      { props.endIndex != props.listLength &&
        <Col smOffset={4} sm={4} xs={12} className="text-center">
          <button className={styles.loadButton} onClick={props.addVideos}>Load More</button>
        </Col>
      }
      <Col sm={4} xs={12} className="text-right">
        <button onClick={props.goToTop} className={styles.upButton}><i className={styles.arrow}></i></button>
      </Col>
    </Row>
  )
}

export default LoadButtonsComponent