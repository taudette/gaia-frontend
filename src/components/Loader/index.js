import React from 'react';
import loader from '../../styles/components/_loader.scss';

const LoaderComponent = () => (
  <div className={loader.loaderContainer}>
    <div className={loader.loader}></div>
  </div>
)

export default LoaderComponent