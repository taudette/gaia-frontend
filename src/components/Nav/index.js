import React from 'react';

export const NavComponent = () => (
  <div>
    Nav Component
  </div>
)

export default NavComponent
