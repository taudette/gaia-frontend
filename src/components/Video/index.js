import React from 'react';
import styles from '../../styles/components/_video.scss'

export const VideoComponent = (props) => {
  console.log(props)
  return (
      <div>
        {/* only render video if image and title are avaiable, check the rest individually */}
        { props.img && props.title && 
          <div className={styles.tileContainer}>
            <img src={props.img} alt="video-image" /> 
            <p className="">{props.title}</p> 
            { props.segment && <p>segment: {props.segment}</p> }
            { props.likes && <p>likes: {props.likes}</p> }
            { props.dislikes && <p>dislikes: {props.dislikes}</p> }
            { props.seasons && <p>seasons: {props.seasons} </p> } 
            { props.episodes && <p>episodes: {props.episodes} </p> } 
          </div>
        }
      </div>
      
  )
}

export default VideoComponent
