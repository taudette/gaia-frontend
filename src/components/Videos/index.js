import React from 'react';
import VideoComponent from '../Video'
import styles from '../../styles/components/_videos.scss';

export const VideosComponent = (props) => {
  if (props.loading) {
    return <p>Loading...</p>
  }

  return (
    <div >
      {props.videos &&
        <ul className={styles.videoList}>
          {props.videos.map((video, index) => {
            return <VideoComponent {...video} key={index} />
          })}
        </ul>
      }
    </div>
  )
}

export default VideosComponent
