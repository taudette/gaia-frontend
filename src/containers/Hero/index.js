import React from 'react';

export const HeroContainer = () => (
  <div>
    Hero
  </div>
)

export default HeroContainer
