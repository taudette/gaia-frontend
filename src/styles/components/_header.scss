@import '../base/_settings.scss';

header {
  padding-left: 4rem;
  padding-right: 4rem;
  display: flex;
  align-items: center;
  height: 5rem;
}

.row {
  width: 100%;
}

.logoContainer {
  width: 7rem;
}

.userContainer {
  float: right;
  text-align: right;

  img {
    width: 2rem;
    margin-left: 1rem;
  }

  span {
    color: darkgrey;
  }
}

.icon {
  background-image: url("../../../public/images/header-logo.svg");
}

