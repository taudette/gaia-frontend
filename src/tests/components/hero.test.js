import React from 'react';
import { shallow } from 'enzyme';
import { HeroComponent } from '../../components/Hero';
import heroData from '../fixtures/hero';

test('should render hero', () => {
  const wrapper = shallow(<HeroComponent hero={heroData} />)
  expect(wrapper).toMatchSnapshot();
})

test('should render hero component without hero data', () => {
  const wrapper = shallow(<HeroComponent hero={{}} />)
  expect(wrapper).toMatchSnapshot();
})