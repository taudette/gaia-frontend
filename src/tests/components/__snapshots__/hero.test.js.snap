// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render hero 1`] = `
<section>
  <div
    className="heroContainer"
  >
    <img
      alt="hero-image"
      src="https://img.gtvcdn.com/cdn/farfuture/Ei4zd5AS7sPaW63_i1saYcvPZkjutLGuv-euzKUStRI/mtime%3A1461683420/sites/default/files/imagecache/term_hero_1440x300/term_images/metaphysics_2.jpg"
    />
    <div
      className="text"
    >
      <h2>
        Metaphysics
      </h2>
      <p
        className="hidden-xs"
      >
        <small>
          There's more to our universe than meets the eye. In this thought-provoking video collection, you'll discover stories of scientific exploration, spiritual metaphysics, alternate realities and mystic messages. Gaia has scoured the globe to bring you the best in metaphysical videos that encourage you to expand your horizons and think outside the box.
        </small>
      </p>
    </div>
  </div>
</section>
`;

exports[`should render hero component without hero data 1`] = `
<section>
  <div
    className="heroContainer"
  >
    <img
      alt="hero-image"
    />
    <div
      className="text"
    >
      <h2 />
      <p
        className="hidden-xs"
      >
        <small />
      </p>
    </div>
  </div>
</section>
`;
